### PR TITLE
Add unit test workflow

### PR DESCRIPTION
**Pull request by f-alizada**
Add unit test workflow to run against pr to main branch.

### CODE REVIEW

File: .github/workflows/unitest.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Unit test
```
Comment (harunrst) [inline .github/workflows/unitest.yml:1]: it doesn't specifically seem like unit tests. we are running all test files in this workflow, let's rename it as just Test or specify unit tests with its folder or however we store it

Comment (f-alizada) [reply .github/workflows/unitest.yml:1]: It runs all .spec files in the src folder, for e2e test I believe there is different folder, having that it runs unit test? Or Am I missing something?

Comment (f-alizada) [reply .github/workflows/unitest.yml:1]: Please Feel free to reopen the comment if I missed something, 
Thank you

Comment (f-alizada) [reply .github/workflows/unitest.yml]: Got it, thanks for review, updated

**Review by harunrst — APPROVED** [review]

**Pull request merged**